docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+
+
+We can reshape the catalog resource to more easily express them, secure them, and claim them.
+
+# Goals
+ 1.  cluster-admin should control which catalog exist.
+ 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
+ 3.  project-admin should be able to delegate per-catalog exposure.
+ 4.  project-editor should be able to claim an entry from the catalog.
+ 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
+ 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
+ 7.  ResourceCatalogEntries should not be restricted to specific claim realization mechanism.
+
+To achieve the ACL related goals, we can create namespaced resources that are selected by label at a cluster level.
+
+
+# API Objects
+
+## ResourceCatalog
+ResourceCatalog is a cluster-scoped resource.  It has a cluster-admin controlled spec that is used to drive a controller.
+The spec includes a label selector that is used in a list/watch to filter all ResourceCatalogEntries in the cluster.
+The controller will find matching entries and copy a reference into the ResourceCatalog.Status field.  The controller
+should have a plug-point for deciding if a given entry is accepted into a given catalog: `Accept(catalogName, entryNamespace, entryName)`
+is probably sufficient.
+
+```go
+// ResourceCatalog is a cluster-scoped resource with a specified spec and a controller maintained status
+type ResourceCatalog struct {
+	unversioned.TypeMeta
+	api.ObjectMeta
+
+	Spec   ResourceCatalogSpec
+	Status ResourceCatalogStatus
+}
+
+type ResourceCatalogSpec struct {
+	// Selector is the selector to use when gathering the namespace scope `ResourceCatalogEntries` for
+	// the `Status`
+	Selector map[string]string
+}
+
+type ResourceCatalogStatus struct {
+	// Entries is dynamically retrieved by selecting all matching ResourceCatalogEntries from all namespaces
+	// This can be done by a controller using a list/watch on all ResourceCatalogEntries based on the spec.selector.
+	Entries []ResourceCatalogInfo
```

Comment:
I'd rather see the list of entries as a virtual resource and not part of the catalog